src/DataDictionary/search/DictionarySearcher/searchHelper.test.js

```diff
@@ -0,0 +1,84 @@
+import {
+  prepareSearchData,
+  searchKeyword,
+  getSearchSummary,
+  ERR_KEYWORD_TOO_SHORT,
+  ERR_KEYWORD_TOO_LONG,
+} from './searchHelper';
+
+describe('dictionary search helper', () => {
+  const dictionary = {
+    a1: {
+      title: 'test',
+      id: 'a1',
+      category: 'A',
+      description: 'whatever',
+      properties: [
+        {
+          name: 'test',
+          description: 'test description',
+          type: 'test type',
+        },
+      ],
+    },
+    a2: {
+      title: 'a2',
+      id: 'a2',
+      category: 'A',
+      description: 'whatever',
+      properties: [],
+    },
+    b1: {
+      title: 'b1',
+      id: 'b1',
+      category: 'B',
+      description: 'whatever',
+      properties: [],
+    },
+    b2: {
+      title: 'b2',
+      id: 'b2',
+      category: 'B',
+      description: 'whatever',
+      properties: [],
+    },
+    b3: {
+      title: 'b3',
+      id: 'b3',
+      category: 'B',
+      description: 'whatever',
+      properties: [],
+    },
+    b4: {
+      title: 'b4',
+      id: 'b4',
+      category: 'B',
+      description: 'whatever',
+      properties: [],
+    },
+  };
+  const searchData = prepareSearchData(dictionary);
+
+  it('can search', () => {
+    const { result } = searchKeyword(searchData, 'test');
+    expect(result.length).toBeGreaterThan(0);
+    const summary = getSearchSummary(result);
+    expect(summary.matchedPropertiesCount).toBeGreaterThan(0);
```

Comment:
I think this test should be more descriptive - it's just checking that `a1` appears in the results but it doesn't check that `a1` is the only result

Reply:
Yeah good idea. I will change to 1. I think here the goal of unit test is to make sure the search library is correctly setup. Whether the search result is correct is out of scope of this unit test, and is already done by search library itself.